### PR TITLE
Fix misspelled keyword argument `poll_interval` for method `acquire`

### LINKED ISSUE #62: poll_intervall is a misspelling
poll_intervall should be spelled poll_interval (but leave the old one for compatibility!)

### PR DESCRIPTION
**Pull request by XuehaiPan**
Fix misspelled keyword argument `poll_interval` for method `acquire`. Fixes #62.

I tried my best to add the backward compatibility layer in this PR to not break existing codebases.

### CODE REVIEW

File: src/filelock/_api.py [hunk 1]
```diff
@@ -106,13 +106,13 @@ def is_locked(self) -> bool:
         """
         return self._lock_file_fd is not None
 
-    def acquire(self, timeout: Optional[float] = None, poll_intervall: float = 0.05) -> AcquireReturnProxy:
+    def acquire(self, timeout: Optional[float] = None, poll_interval: float = 0.05, **kwargs) -> AcquireReturnProxy:
```
Comment (gaborbernat) [inline src/filelock/_api.py]: Please don't accept everything, the only thing we do need to accept is the old variant.

File: src/filelock/_api.py [hunk 2]
```diff
@@ -140,6 +140,17 @@ def acquire(self, timeout: Optional[float] = None, poll_intervall: float = 0.05)
         if timeout is None:
             timeout = self.timeout
 
+        # TODO: remove this and `**kwargs` above in a future release
```
Comment (gaborbernat) [inline src/filelock/_api.py]: No TODO please

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/filelock/_api.py (hunk 1 above)
Comment (XuehaiPan) [reply src/filelock/_api.py]: At lines 151-152, it will raise `TypeError` when the user provides something other than the old variant.

Comment (gaborbernat) [reply src/filelock/_api.py]: Yeah, I saw. But we don't need that, just express the intent via the interface, not fixup code.

File: src/filelock/_api.py [hunk 3]
```diff
@@ -140,6 +140,17 @@ def acquire(self, timeout: Optional[float] = None, poll_intervall: float = 0.05)
         if timeout is None:
             timeout = self.timeout
 
+        # Remove this and `**kwargs` above in a future release
+        if "poll_intervall" in kwargs:
+            _LOGGER.warning(
```
Comment (gaborbernat) [inline src/filelock/_api.py]: This should be done via the warnings module

File: src/filelock/_api.py [hunk 4]
```diff
@@ -106,13 +107,18 @@ def is_locked(self) -> bool:
         """
         return self._lock_file_fd is not None
 
-    def acquire(self, timeout: Optional[float] = None, poll_intervall: float = 0.05) -> AcquireReturnProxy:
+    def acquire(
+        self,
+        timeout: Optional[float] = None,
+        poll_interval: float = 0.05,
```
Comment (gaborbernat) [inline src/filelock/_api.py:113]: Why are you changing the interface type here?

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Can you please add a test to validate this?

**Comment by XuehaiPan** [comment]
I think we can reimplement this by using decorators without changing the signature of the method.

**Comment by gaborbernat** [comment]
> I think we can reimplement this by using decorators without changing the signature of the method.

I think the current implementation is fine, no need for decorators.

**Review by gaborbernat — APPROVED** [review]

**Comment by gaborbernat** [comment]
Thanks for your contribution.

**Comment by ofek** [comment]
I don't know where it's coming from (maybe virtualenv?) but I'm getting a lot of warnings:

```
tests/cli/build/test_build.py: 10 warnings
tests/cli/env/test_create.py: 45 warnings
tests/cli/env/test_prune.py: 6 warnings
tests/cli/env/test_remove.py: 12 warnings
tests/cli/publish/test_publish.py: 42 warnings
tests/cli/run/test_run.py: 18 warnings
tests/dep/test_core.py: 24 warnings
tests/venv/test_core.py: 21 warnings
  /home/runner/.local/share/hatch/env/virtual/hatch-kmtIAcBK/hatch/lib/python3.9/site-packages/filelock/_api.py:151: DeprecationWarning: use poll_interval instead of poll_intervall
    warnings.warn(msg, DeprecationWarning)
```

**Comment by XuehaiPan** [comment]
> I don't know where it's coming from (maybe virtualenv?) but I'm getting a lot of warnings:

@ofek Can you try to run `python -W error xxx.py` or `pytest ... -W error` to see what module uses the old argument `poll_intervall`?

**Comment by ofek** [comment]
Thanks, found it!

https://github.com/pypa/virtualenv/blob/623189a00aa96c805990fd12dc883c69e251f4b2/src/virtualenv/util/lock.py#L31

**Comment by ofek** [comment]
https://github.com/pypa/virtualenv/pull/2237

**Pull request merged**